fix(crashlytics): resolve ReEnterPinActivity crashes and deprecation warnings

JosephSanjaya opened the pull request:
## 📱 Description
Fixes critical ReEnterPinActivity crashes and modernizes the PIN entry system by migrating from legacy Java + XML to Kotlin + Compose. This PR resolves NullPointerException crashes, addresses deprecation warnings, and implements a complete modern PIN re-entry flow with proper error handling and user feedback.

## Platform
- [x] Android
- [ ] iOS
- [ ] Games-Unity
- [ ] DevOps (AWS)
- [ ] Website
- [ ] C/Golang 

## 🎯 Type of Change
- [x] 🐛 Bug fix
- [x] ✨ New feature
- [x] 🔧 Refactoring (code change that neither fixes a bug nor adds a feature)
- [ ] 📚 Documentation update
- [x] 🎨 UI/UX improvement
- [ ] ⚡ Performance improvement
- [x] 🧪 Test addition or improvement

## 📋 Changes

### 🐛 Critical Bug Fixes
- **Add missing VIBRATE permission** to prevent SecurityException crashes during haptic feedback
- **Fix API compatibility issues** with activity transitions (supports API 29+ with proper fallbacks)
- **Resolve NullPointerException crashes** by implementing proper null safety checks for intent extras
- **Remove unnecessary SDK version checks** for vibration (min SDK is 29, so API checks were redundant)

### 🔧 Modernization & Refactoring
- **Replace deprecated `onBackPressed`** with modern `OnBackPressedDispatcher` for proper back navigation handling
- **Update activity transitions** to use `overrideActivityTransition` (API 34+) with fallback to deprecated method for older versions
- **Migrate from Java + XML to Kotlin + Compose + MVI** architecture using Orbit pattern
- **Implement comprehensive state management** with proper error handling and user feedback
- **Complete Compose UI implementation** with modern Material Design 3 components
- **Reusable PIN input components** (`PinDot`, `PinKeyboard`, `PinInputComponent`) for future use
- **Proper theming support** with light/dark mode compatibility
- **Enhanced user experience** with shake animations, haptic feedback, and visual error states
- **Comprehensive unit tests** with 100% coverage of ViewModel logic

### 🧹 Cleanup & Integration
- **Remove legacy Java ReEnterPinActivity** and associated XML layouts
- **Update SetPinActivity integration** to use new `ReEnterPinActivity.createIntent()` method
- **Clean up AndroidManifest references** and update activity declarations
- **Remove obsolete references** in BreadActivity to deleted legacy activity

### 🔗 Related Issues
- Crashlytics: https://console.firebase.google.com/project/brainwallet-mobile/crashlytics/app/android:ltd.grunt.brainwallet/issues/586aca639a3014bd77cebaff2d7b4602?time=90d&types=crash&sessionEventKey=68EB8492033E00013FD9795EF6106EDF_2138727769913111126
- Fixes https://github.com/gruntsoftware/internal/issues/271

### 🧪 Tests Status
- [x] Tests ran successfully locally?
- [x] Added more tests? How many? **15 comprehensive unit tests**
- [x] Code coverage percentage of the codebase: **100%** (for new ViewModel logic)

## 📸 Screenshots/Videos
![Screen_recording_20251020_011327](https://github.com/user-attachments/assets/487f946e-0b2e-4194-b258-530e6bca19ff)


## 🎯 Reviewers
@kcw-grunt, @josikienow

---

JosephSanjaya committed:
fix: resolve ReEnterPinActivity crashes and deprecation warnings

- Add missing VIBRATE permission to prevent SecurityException crashes
- Fix API compatibility issues with activity transitions (API 29+ support)
- Remove unnecessary SDK version checks for vibration (min SDK is 29)

refactor: modernize ReEnterPinActivity navigation handling

- Replace deprecated onBackPressed with OnBackPressedDispatcher
- Update activity transitions to use modern overrideActivityTransition (API 34+) with fallback
- Update SetPinActivity to use new ReEnterPinActivity.createIntent() method
- Remove legacy Java ReEnterPinActivity and update AndroidManifest references
- Clean up BreadActivity references to removed legacy activity

josikie commented:
👀

kcw-grunt commented:
Me too 👀

kcw-grunt commented on app/src/main/AndroidManifest.xml:
```diff
@@ -11,6 +11,7 @@
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.BIND_JOB_SERVICE" />
+    <uses-permission android:name="android.permission.VIBRATE" />
```
Curious if this is an opportuntiy @JosephSanjaya .

Glad it was caught @JosephSanjaya

kcw-grunt commented on app/src/main/java/com/brainwallet/presenter/activities/BreadActivity.java:
```diff
@@ -135,7 +135,7 @@ protected void onCreate(Bundle savedInstanceState) {
         secondaryPrice.setTextSize(SECONDARY_TEXT_SIZE);
 
-        finishActivities(SetPinActivity.introSetPitActivity, ReEnterPinActivity.reEnterPinActivity);
+        finishActivities(SetPinActivity.introSetPitActivity, null);
```
Why @JosephSanjaya ?

josikie replied:
Why is it null?

kcw-grunt commented on app/src/main/java/com/brainwallet/ui/composable/PinKeyboard.kt:
```diff
@@ -0,0 +1,150 @@
+package com.brainwallet.ui.composable
+
+import android.content.res.Configuration
```
Did you go through the process of starting from scratch @JosephSanjaya  This way we know we dont get surprises

JosephSanjaya replied:
I will try to sanity it again

kcw-grunt reviewed (dismissed):
Approved.
Please merge if the New path was tested @JosephSanjaya

JosephSanjaya commented:
I check the happy path, from createing and restoring wallet, it working fine, is there any flow i need to check @kcw-grunt ?

josikie reviewed (dismissed):
Good, thanks! @JosephSanjaya

JosephSanjaya committed:
Refactor: Remove legacy PIN activities and layouts

This commit removes the legacy PIN management activities (`SetPinActivity`, `ReEnterPinActivity`, `UpdatePinActivity`) and their associated layouts and tests. The functionality for setting, re-entering, and updating the PIN has been consolidated and moved into a modern, composable architecture.

Specific changes include:
- Deletion of `SetPinActivity`, `UpdatePinActivity`, and their XML layout `activity_pin_template.xml`.
- Deletion of the Kotlin-based `ReEnterPinActivity` and its associated MVI components (Intent, State, SideEffect, ViewModel) and unit tests.
- Removal of references to these deleted activities from `AndroidManifest.xml` and other parts of the codebase like `PromptManager` and `SecurityCenterActivity`.
- Update of `tools:context` attributes in several XML layouts that were incorrectly pointing to `SetPinActivity`.
- `PromptManager` and `SecurityCenterActivity` are updated to navigate to the new `BrainwalletActivity` with an `UnLock` route for PIN-related actions.
- Cleaned up related code in `BreadActivity` and `ActivityUTILS`.

JosephSanjaya commented:
Hello @kcw-grunt , @josikie really sorry turn out this code already not used thats why it just work without noticing changes, i remove legacy code and add sanity video. can you recheck

kcw-grunt reviewed (approved):
LGTM>

@JosephSanjaya  feel free to **Squash and Merge**
